### ISSUE TITLE
[Bug]: tpcc consistency check error when run tpcc test

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#5285]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):0.6.0
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

after run tpcc, consistency check error:
mysql> (Select w_id, w_ytd from bmsql_warehouse) except(select d_w_id, sum(d_ytd) from bmsql_district group by d_w_id); 
+------+-----------+
| w_id | w_ytd     |
+------+-----------+
|    1 | 314076.00 |
+------+-----------+
1 row in set (0.10 sec)

The reason is that:
wid=1,did=9,dytd=34418.60,incr=4418.59
wid=1,wytd=304419.00,incr=4418.59
wid=1,did=1,dytd=33302.70,incr=3302.74
wid=1,wytd=307722.00,incr=3302.74
wid=1,did=4,dytd=30772.30,incr=772.29
wid=1,wytd=308494.00,incr=772.29
wid=1,did=3,dytd=32443.50,incr=2443.49
wid=1,wytd=310937.00,incr=2443.49
wid=1,did=8,dytd=33138.80,incr=3138.82
wid=1,wytd=314076.00,incr=3138.82

when decimal type increase a value, the precision is lost, such as wid=1,did=9,dytd=34418.60,incr=4418.59,
the old value of dytd is 300000, when increase 4418.59, the result is 34418.60

java code is :

stmt = db.stmtPaymentUpdateDistrict;
stmt.setDouble(1, payment.h_amount);// the decimal value is set by setDouble
stmt.setInt(2, payment.w_id);
stmt.setInt(3, payment.d_id);
stmt.executeUpdate();

**Comment by aressu1985** [comment]
this probem still exsists
commit:554cca326583c75b4c438c2888345e810c76b655

mysql> (Select w_id, w_ytd from bmsql_warehouse) except(select d_w_id, sum(d_ytd) from bmsql_district group by d_w_id); 
+------+-----------+
| w_id | w_ytd     |
+------+-----------+
|    1 | 461370.91 |
+------+-----------+
1 row in set (0.17 sec)

**Comment by aressu1985** [comment]
2022-11-02 16:49:40 INFO  jTPCCTerminal:350 - Term-01, Executing for a limited time...
wid=1,did=8,dytd=31209.94,incr=1209.94
wid=1,wytd=301209.94, sum_d_ytd=301209.94, incr=1209.94
wid=1,did=5,dytd=30724.33,incr=724.33
wid=1,wytd=301934.27, sum_d_ytd=301934.27, incr=724.33
wid=1,did=4,dytd=34789.60,incr=4789.6
wid=1,wytd=306723.90, sum_d_ytd=306723.87, incr=4789.6
wid=1,did=2,dytd=34859.85,incr=4859.85
wid=1,wytd=311583.75, sum_d_ytd=311583.72, incr=4859.85
wid=1,did=8,dytd=33000.20,incr=1790.26
wid=1,wytd=313374.01, sum_d_ytd=313373.98, incr=1790.26
wid=1,did=5,dytd=35178.90,incr=4454.6
wid=1,wytd=317828.60, sum_d_ytd=317828.55, incr=4454.6
wid=1,did=9,dytd=34934.07,incr=4934.07
wid=1,wytd=322762.67, sum_d_ytd=322762.62, incr=4934.07
wid=1,did=5,dytd=40123.92,incr=4945.02
wid=1,wytd=327707.69, sum_d_ytd=327707.64, incr=4945.02
wid=1,did=1,dytd=32834.31,incr=2834.31
wid=1,wytd=330542.00, sum_d_ytd=330541.95, incr=2834.31

**Comment by ouyuanning** [comment]
create table t1 (a decimal(12,2), b decimal(12, 2));
insert into t1 values (301934.27, 301934.27);
update t1 set a = a + 4789.6, b = b + 4789.60;
select * from t1;

**Comment by aressu1985** [comment]
fixed e23e00f1d01d43e33c509a6d239ccc38b5f93409
